test/test_mahalanobis_mixin.py

```diff
@@ -272,26 +272,25 @@ def test_get_squared_metric(estimator, build_dataset):
                          ids=ids_metric_learners)
 def test_transformer_is_2D(estimator, build_dataset):
   """Tests that the transformer of metric learners is 2D"""
-  # TODO: remove this check when SDML has become robust to 1D elements,
-  #  or when the 1D case is dealt with separately
-  if not str(estimator).startswith('SDML'):
-    input_data, labels, _, X = build_dataset()
-    model = clone(estimator)
-    set_random_state(model)
-    # test that it works for X.shape[1] features
-    model.fit(input_data, labels)
-    assert model.transformer_.shape == (X.shape[1], X.shape[1])
-
-    # test that it works for 1 feature
-    trunc_data = input_data[..., :1]
-    # we drop duplicates that might have been formed, i.e. of the form
-    # aabc or abcc or aabb for quadruplets, and aa for pairs.
-    slices = {4: [slice(0, 2), slice(2, 4)], 2: [slice(0, 2)]}
-    if trunc_data.ndim == 3:
-      for slice_idx in slices[trunc_data.shape[1]]:
-        _, indices = np.unique(trunc_data[:, slice_idx, :], axis=2,
-                               return_index=True)
-        trunc_data = trunc_data[indices]
-        labels = labels[indices]
-    model.fit(trunc_data, labels)
-    assert model.transformer_.shape == (1, 1)  # the transformer must be 2D
+  input_data, labels, _, X = build_dataset()
+  model = clone(estimator)
+  if model.__class__.__name__.startswith('SDML'):
```

Comment:
this is a special check that in the case where we have SDML, we ensure we are in the case where the matrix to reconstruct is PSD. This is a specific check but it's done sometimes in scikit-learn, (e.g. transforming the data into a square matrix for algorithms with kernel='precomputed', and I could find more). It's ok, because the purpose of this test is just to test the form (2D) of the output in some standard case. Corner cases behaviours shoudl be (and are) dealt with separately (in this PR we decide to raise a warning in the case where the matrix to reconstruct is not PSD). 
We could also ensure to have one good form of the input good for all estimators, but first it's a bit more complicated (we would need to find a reasonable (to work with all alrogithms) dataset with 1 feature such as the matrix is PSD), and if we add an algo that fails we have to rethink the dataset etc... So it's better (and simpler) to have a reasonable dataset (extracted from iris), and to tune the algorithms not to fail on it, since we just want to check they return a 2d metric in the case of 1d input here